### ISSUE TITLE
Current 6.0 SDK building in VS errors with 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.

### ISSUE THREAD

**Issue by benaadams** [dotnet/sdk#17461]
In VS building a mostly empty web project fails with 
```
1>------ Rebuild All started: Project: WebApplication1, Configuration: Release Any CPU ------
1>C:\Program Files\dotnet\sdk\6.0.100-preview.5.21255.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props(14,3):
 error : C:\Program Files\dotnet\sdk\6.0.100-preview.5.21255.14\Sdks\Microsoft.NET.SDK.WorkloadAutoImportPropsLocator\Sdk not found.
 Check that a recent enough .NET SDK is installed and/or increase the version specified in global.json.

1>C:\Program Files\dotnet\sdk\6.0.100-preview.5.21255.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props(14,38):
 error MSB4236: The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.

1>Done building project "WebApplication1.csproj" -- FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```
At CLI it suceeds by has a different warning
```
Microsoft (R) Build Engine version 16.11.0-preview-21254-21+e73d08c28 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Users\thund\source\repos\WebApplication1\WebApplication1\WebApplication1.csproj (in 62 ms).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview

C:\Program Files\dotnet\sdk\6.0.100-preview.5.21255.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5):
 warning NETSDK1048: 'AdditionalProbingPaths' were specified for GenerateRuntimeConfigurationFiles,
 but are being skipped because 'RuntimeConfigDevPath' is empty. 

[C:\Users\thund\source\repos\WebApplication1\WebApplication1\WebApplication1.csproj]
  WebApplication1 -> C:\Users\thund\source\repos\WebApplication1\WebApplication1\bin\Debug\net6.0\win-x64\WebApplication1.dll

Build succeeded.
```

**Comment by benaadams** [comment]
Causes **all** VS builds to fail including net5.0 ones. 

Adding `global.json` does allow them to build
```json
{
    "sdk": {
        "version": "5.0.202",
        "rollForward": "latestFeature"
    }
}
```

**Comment by davidfowl** [comment]
I'm hitting this too. I reverted my upgrade because of this

**Comment by dsplaisted** [comment]
The .NET 6 SDK now enables [workload resolvers](https://github.com/dotnet/designs/blob/main/accepted/2020/workloads/workload-resolvers.md) by default.  This means when using Full Framework MSBuild (which happens in Visual Studio or when you use `msbuild` instead of `dotnet` from the command line), you need the workload resolver enabled.  This is enabled by default in VS 16.10 Preview 3.  For previous versions of MSBuild, you can set the `MSBuildEnableWorkloadResolver` environment variable to `true`.  However, this workaround is likely to stop working in Preview 4 or Preview 5 of .NET SDK 6.0.100, as there is a change to the workload manifest schema which will cause the previous version of the resolver to fail.

In short: Update to VS 16.10 Preview 3 or later if you can, and if you can't you can try setting the `MSBuildEnableWorkloadResolver` environment variable to `true`.

**Comment by davidfowl** [comment]
I tried updating both VS and the .NET SDK and ASP.NET Core basically disappeared. I can't seem to make a new project (or open an existing one).

**Comment by davidfowl** [comment]
Actually, I can't seem to create any projects...

**Comment by davidfowl** [comment]
Repaired VS, still doesn't work. I'm gonna declare that there's a bug here.

**Comment by benaadams** [comment]
> Actually, I can't seem to create any projects...

I get this when doing new project in VS https://github.com/dotnet/sdk/issues/17515 have to create a new one in cli

**Comment by dsplaisted** [comment]
Actually, a better workaround is probably to set `MSBuildEnableWorkloadResolver` to false.  This should avoid the issue with the manifest schema change.

Of course, eventually after you are using an updated version of VS you'll want to unset the environment variable to allow workloads to work.

**Comment by davidfowl** [comment]
I'm using internal dogfood builds and its still broken in the ways expressed here.

**Comment by akamud** [comment]
I'm having this same issue on macOS, either on Rider or running `dotnet build` from the command line. .NET 6 was working a few days ago when I was using the nightly releases, stopped working when I installed Preview 4.

**Comment by michaelherndon** [comment]
This issue is definitely causing problems with omnisharp.  No luck with the MSBuildEnableWorkloadResolver environment variable. Uninstalling the preview caused even more issues.  I had to reinstall the preview and install the preview version of Visual Studio.

**Comment by zivkan** [comment]
> Actually, a better workaround is probably to set MSBuildEnableWorkloadResolver to false. This should avoid the issue with the manifest schema change.

Can the SDK make this feature opt-in, rather than opt-out, until it's ready enough to work without such mitigations?

NuGet has a bunch of functional tests that restore various projects, and it's a bit of a burden to find all the places we need to set this. Plus we'll need to remember to remove it again in the future.

**Comment by Webreaper** [comment]
Seeing the same thing with .Net 6-preview-5. If I do:
```
export MSBuildEnableWorkloadResolver=false
```
that works around it and lets me build.

**Comment by slang25** [comment]
For those not wanting to remember to set this env var, you can add a `Directory.Build.rsp` file in the root of your repo to achieve the same thing, then once this gets fixed you can delete the file.

Example: https://github.com/slang25/JustEat.StatsD/commit/30fffb3c5cf6d94fb286642220445a95df5aa2fc

**Comment by Webreaper** [comment]
Just upgraded to VS for Mac 8.10.6 build 10, and .Net 6.0 preview-6, and still getting this. Any idea when it's likely to be fixed?

**Comment by RussKie** [comment]
Just hit this too
```
C:\Program Files\dotnet\sdk\6.0.100-preview.7.21358.10\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props(14,3): warning MSB4242: The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed to run. An item with the same key has already been added. Key: Microsoft.NET.Runtime.Emscripten.Node
C:\Program Files\dotnet\sdk\6.0.100-preview.7.21358.10\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props(14,38): error MSB4236: The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found. [C:\Development\DesignerTestHarness\WindowsFormsApp472\NetFx472.csproj]
```

**Comment by RussKie** [comment]
@dsplaisted looks like I can't use VS at all, outside projects that use local version of .dotnet...

**Comment by davidfowl** [comment]
When this happens I typically rename the `C:\Program Files\dotnet\sdk-manifests` folder, then restart VS

**Comment by RussKie** [comment]
> When this happens I typically rename the `C:\Program Files\dotnet\sdk-manifests` folder, then restart VS

Thanks! That worked.

**Comment by davidwengier** [comment]
I just set the `MSBuildEnableWorkloadResolver` environment variable on all my machines to `false`. Looking forward to what breaks in future when I forget I've done that 😀

**Comment by dsplaisted** [comment]
I believe that with #18166 these error messages will get a bit more useful.

**Comment by marcpopMSFT** [comment]
I think the more recent issues described by Russ happens when you install preview 7 and then install VS (which has preview 6), you get mismatching manifests with emscripten and it breaks.  You can repair preview 7 to work around this or delete the manifests folder as you've done. It'll get resolved once we put preview 7 in VS.

**Comment by CyrusNajmabadi** [comment]
So i've hit this as well.  It tooks us almost a week to work through this.  The only thing that worked was setting `MSBuildEnableWorkloadResolver` to `false`.  Can fixing this be prioritized?  (tagging @marcpopMSFT ).  Thanks!

**Comment by marcpopMSFT** [comment]
The fix requires either getting preview 7 inserted into VS or going back in time and not breaking compat between preview 6 and 7. Unfortunately, the .Net build system is not conducive to created earlier signed builds and inserting to VS so typically this happens very late in each milestone.  The good news is that this should be the final compat break in the workloads and so preview 7 through GA should not have the same issues.  

This compat issue was communicated to tactics but we could have gone more broadly with it.

**Comment by davidfowl** [comment]
Feels like this has been broken for 3 previews but I'm happy it's getting fixed in preview 7

**Comment by marcpopMSFT** [comment]
There has been a lot of churn in the workloads (which we expected) but also churn in their names and dependencies (less expected).  It should be fixed with preview 7 going forward mostly because the workload owners are done with that level of breaking change.  We have a design for how to allow for smooth name changes with aliasing between different workloads but that work didn't get done for .net 6 (and was going to be a breaking change all on it's own).

**Comment by Webreaper** [comment]
Installed .Net 6 preview 7 this morning, and I still get the error about this workload when I launch Visual Studio for Mac, so still have to set `export MSBuildEnableWorkloadResolver=false` to be able to continue working. So I'm not sure this is fixed in preview 7.

**Comment by slang25** [comment]
I fixed it by deleting the following folders from `/usr/local/share/dotnet/sdk-manifests/6.0.100`
![image](https://user-images.githubusercontent.com/1341446/129397224-b30bd1f8-1291-4c2a-8247-8d44a4b2a348.png)

I believe one of the older previews put these folders there, so it could be that it's fixed from occurring again, but with no automatic fix-up shipping in new previews.

**Comment by Webreaper** [comment]
Nice! Will try that later.

I had some problems with preview 7, and when I reinstalled preview 6 they continued - in the end I had to manually delete/uninstall the preview 7 files that had been left behind. It might be useful if the folks from MSFT could add an option in the SDK installer that force-deleted all other older (and newer) previews completely, so people could ensure they were running with a clean install. Might reduce the number of false positive bug reports.

**Comment by marcpopMSFT** [comment]
@sfoslund for visibility as we tried to put a hack into the resolver to avoid this particular rename but it looks like that may have only applied to VS. Workloads were designed to be SxS across different feature bands but not within the same one which previews are.

**Comment by Webreaper** [comment]
> I fixed it by deleting the following folders from `/usr/local/share/dotnet/sdk-manifests/6.0.100`

I tried deleting those folders, but it didn't fix the issue for me. I'm using the VS4Mac 2022 preview, and this workload issue hangs it at startup so you can't quit - the only way to solve it is force-quit. Luckily the `export MSBuildEnableWorkloadResolver=false` solves it in there too.

**Comment by MichaelSL** [comment]
```
The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.  C:\Program Files\dotnet\sdk\6.0.100-preview.7.21379.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props```

Setting env variable helps.

Could someone confirm that this doesn't work with `preview7` for them too?

**Comment by TaylorWatson** [comment]
Nuking the entire `/usr/local/share/dotnet/sdk-manifests/6.0.100` from orbit seems to have resolved my issues -.- This has been such a pain.

**Comment by Webreaper** [comment]
> Nuking the entire `/usr/local/share/dotnet/sdk-manifests/6.0.100` from orbit seems to have resolved my issues -.- This has been such a pain.

That did the trick. Thanks!

**Comment by mhutch** [comment]
For folks seeing this error, here's a summary/overview.

The error `The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found` can be caused by several underlying issues.

It may indicate that the .NET workload resolver is not present in the MSBuild directory. This happens with versions of Visual Studio that do not support the .NET 6 previews. It could also happen in corrupt or highly customized MSBuild installations.

Otherwise, it indicates an internal error in the workload resolver, most likely when reading the workload manifests in the .NET SDK. A known case is a composition error due to stale manifests from an older .NET 6 preview, which can be fixed by removing the `dotnet/sdk-manifests/6.0.100` directory from the .NET installation and reinstalling the latest .NET 6 SDK preview. If that doesn't fix it, check the MSBuild log for details of the internal error. It will be [surfaced better](https://github.com/dotnet/msbuild/issues/6762) in future.

if you can't find get to an MSBuild log in the IDE, you can invoke msbuild from the CLI. Be aware that Visual Studio includes its own copy of MSBuild, which can be invoked as follows:
* Windows: use the Developer Command Prompt shortcut matching the VS instance, and invoke `msbuild` from that prompt.
* Mac: install [vsmac-cli](https://www.nuget.org/packages/vsmac-cli/0.2.0) and invoke `vsmac msbuild` from Terminal.app. If you have multiple VS instances, see `vsmac --help` for arguments to select a specific instance.

**Comment by slang25** [comment]
I had no idea [vsmac-cli](https://www.nuget.org/packages/vsmac-cli/0.2.0) existed, that's really useful!

**Comment by vova-lantsov-dev** [comment]
I got this issue in a docker container running a clean installation of .NET SDKs.
Here is some info:
**Docker:** 20.x (latest)
**Base image:** balenalib's aarch64 (arm64v8) debian bullseye
**.NET sdk installed in docker container:** 3.1, 5, 6-rc.1
**Installation instructions:** [here](https://github.com/vlrpi/rpi-teamcity-agent-dotnet/blob/main/debian/bullseye/2021.1/dotnet-supported/Dockerfile)
**Logs:** [here](https://tc.vova-lantsov.dev/buildConfiguration/Vlrpi_Build_CompileAndPushTeamcityAgentDotnet/302?buildTab=log) (login as guest)
**Project:** [here](https://github.com/vlrpi/build) (running using Nuke build automation system)
**Server instance:** Raspberry Pi 4 Model B (8 GB RAM)

Contact me if you need more info about my case

**Comment by dotMorten** [comment]
I've been hitting this a lot now too. It fails with both VS2019 and 2022, provided you have .NET6-rc1 installed. The repro steps are pretty simple:
```
git clone https://github.com/Esri/arcgis-toolkit-dotnet
cd arcgis-toolkit-dotnet\src\Toolkit\Toolkit
msbuild /restore Esri.ArcGISRuntime.Toolkit.csproj
```

**Comment by Webreaper** [comment]
@dotMorten Have you got any other earlier .Net 6 previews installed? I saw this frequently until preview-7, but it only went away when I physically deleted all the old previews, per @TaylorWatson's comment previously (see also [here](https://docs.microsoft.com/en-us/dotnet/core/install/remove-runtime-sdk-versions?pivots=os-macos#uninstall-net)).

**Comment by dotMorten** [comment]
@Webreaper One one PC where I reproduce it, it's a completely fresh install. Never had anything but RC1 and VS2022P4 on it. I'm starting to think my issue is WPF specific though.

**Comment by marcpopMSFT** [comment]
Closing this issue out with the release of RC2 which we believe should fix most instances of this.  Folks still seeing issues should try to clean out their sdks (in particular old sdk-manifests) and reinstall as many people still hitting this had leftover files from old maui-check calls. I think dotMorten's repro is a completely separate issue than this.

**Comment by vova-lantsov-dev** [comment]
@marcpopMSFT I still have this issue with .NET 6 rc2, my case is described a few comments above

Here is full log of build: https://tc.vova-lantsov.dev/buildConfiguration/Vlrpi_Build_CreateBuilder/402?buildTab=log (login as guest)

**Comment by vova-lantsov-dev** [comment]
@marcpopMSFT that's a clean installation inside the docker container, after docker cache was pruned.

How I install SDKs: [link](https://github.com/vlrpi/rpi-teamcity-agent-dotnet/blob/main/debian/bullseye/2021.1/dotnet-supported/arm64/Dockerfile)

**Comment by marcpopMSFT** [comment]
@sfoslund To clarify, I think the customer is using arm64 linux docker images, isn't trying to do workloads at all, but is getting workload failures when just using this sdk. I'm wondering if our workload fallback logic is working incorrectly here. I don't think they are doing a workload install at all from the build results. Could be something off specifically with our linux arm64 images maybe or our fallback logic

Here is full log of build: https://tc.vova-lantsov.dev/buildConfiguration/Vlrpi_Build_CreateBuilder/402?buildTab=log (login as guest)

**Comment by vova-lantsov-dev** [comment]
@marcpopMSFT what you mean by workload?

**Comment by marcpopMSFT** [comment]
> @marcpopMSFT what you mean by workload?
.NET 6 introduces a new concept to the SDK called Optional Workloads for acquiring large components that extend the SDK. If you're not doing Maui development or specific Blazor Wasm compilation, you aren't using Optional Workloads. The error you're seeing though comes from our workloads logic so I'm thinking there is something specifically wrong with the arm64+linux image.

**Comment by danwalmsley** [comment]
This is still an issue on the release version, even on Azure pipelines.

see this build for example:
https://dev.azure.com/AvaloniaUI/AvaloniaUI/_build/results?buildId=16510&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=5494183b-801e-5084-3b4b-5451c6a14ee1

**Comment by dotMorten** [comment]
@danwalmsley Really? Mine case is better now.

**Comment by apousada** [comment]
Bumping... I see a similar exception in the log produced under %TEMP%\servicehub\logs\ when trying to start Live Unit Testing:

```
Error : 10 :Exception thrown from request "4" for method LoadProjectAsync: System.AggregateException: One or more errors occurred. ---> Microsoft.Build.Exceptions.InvalidProjectFileException: The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.  C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props
```

Setting the `MSBuildEnableWorkloadResolver` environment variable to `false` seems to get me by for now.

**Comment by marcpopMSFT** [comment]
@sfoslund to look into the LUT report above.

**Comment by sfoslund** [comment]
@apousada can you give a full repro of this issue? 

cc @dsplaisted in case you've seen this issue before or have any ideas.

**Comment by goaaats** [comment]
This happens to us using the `windows-2022` GitHub Actions image. Setting `MSBuildEnableWorkloadResolver` to `true` doesn't seem to fix it.

https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#net-core-sdk
https://github.com/goatcorp/Dalamud/runs/4252003866?check_suite_focus=true#step:5:19

**Comment by hellfirehd** [comment]
This is still happening for me as well. 

* Windows 11 Pro Workstation clean install (not upgrade)
* VS 2022 clean install, no VS 2019
* Nuke 5.3.0
* `<Project Sdk="Microsoft.NET.Sdk">`
* `<TargetFrameworks>net6.0</TargetFrameworks>`
* Don't know what `MSBuildEnableWorkloadResolver` is so I assume it's not set.

```powershell
❯ nuke
PowerShell Desktop version 5.1.22000.282
Microsoft (R) .NET Core SDK version 6.0.100
Creating directory 'C:\Source\Redacted.EntityFrameworkCore\.nuke\temp'...

███╗   ██╗██╗   ██╗██╗  ██╗███████╗
████╗  ██║██║   ██║██║ ██╔╝██╔════╝
██╔██╗ ██║██║   ██║█████╔╝ █████╗
██║╚██╗██║██║   ██║██╔═██╗ ██╔══╝
██║ ╚████║╚██████╔╝██║  ██╗███████╗
╚═╝  ╚═══╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝

NUKE Execution Engine version 5.3.0 (Windows,.NETCoreApp,Version=v5.0)

The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.  C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props
StackTrace:
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpressionConditioned(String directoryOfImportingFile, ProjectImportElement importElement, List`1& projects, SdkResult& sdkResult, Boolean throwOnFileNotExistsError)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImports(String directoryOfImportingFile, ProjectImportElement importElement, SdkResult& sdkResult)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate()
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project..ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.FromProjectRootElement(ProjectRootElement rootElement, ProjectOptions options)
   at Nuke.Common.ProjectModel.ProjectModelTasks.ParseProject(String projectFile, String configuration, String targetFramework)
   at Nuke.Common.Execution.Telemetry.CheckAwareness()
```

**Comment by vova-lantsov-dev** [comment]
@hellfirehd please read #22221 (or better visit official Nuke community in Slack to get support there)

**Comment by apousada** [comment]
> @apousada can you give a full repro of this issue?
> 
> cc @dsplaisted in case you've seen this issue before or have any ideas.

Sorry for the delay! Yes, a barebones solution that reproduces the issue for me is attached. The steps are simply to open Visual Studio 2022, load the solution and try starting Live Unit Testing when the `MSBuildEnableWorkloadResolver` environment variable is either not set or is set to `true`. Close Visual Studio 2022, set the environment variable to `false`, and then re-open VS and the solution, start unit testing, and then it works for me.

[LutWorkloadResolverIssue.zip](https://github.com/dotnet/sdk/files/7588233/LutWorkloadResolverIssue.zip)
.

**Comment by AreoByte** [comment]
I am also having this issue. Attempting to run my project (https://github.com/ppy/osu) in monodevelop gives me:
`/usr/share/dotnet/sdk/6.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.props(38,38): Error MSB4236: The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found. (MSB4236) (osu.Game)`
But if I run it in CLI (dotnet run --project osu.Desktop) it runs just fine. I am not sure if this is an issue relating to dotnet or monodevelop, but I read [here](https://developercommunity.visualstudio.com/t/monodevelopcoreuserexception-unable-to-find-sdk-mi/1459162) that this is apparently an issue with dotnet, so I figured putting it here.

**Comment by AraHaan** [comment]
That is because monodevelop is not supported anymore I think at all.

**Comment by kwaazaar** [comment]
I created a .NET 4 webapplication project and migrated it to the SDK-style projectfile format (using [this project](https://github.com/CZEMacLeod/MSBuild.SDK.SystemWeb)). To build it I need to use msbuild (as both `dotnet build` and `dotnet msbuild` don't work. Msbuild is installed on our custom agents, but does not contain the .NET Core Build Tools workload, as this no longer exists since VS2022. By specifying MSBuildSDKsPath I can make it find the SDKs and I need to set MSBuildEnableWorkloadResolver to false to prevent this error.

**Comment by baskren** [comment]
I would like to ask for this issue to be reopened.  Upon install of VS2022 (v17.0 - build8989) on x64 Mac, I get this error.  Do not get this error when building from command line.  

Have 3x deleted every trace of Mono, Xamarin, NuGet, DotNet, and VisualStudio from my system and reinstalled VS2022 to the same result.

Setting the `MSBuildEnableWorkloadResolver` environment variable does not help.

**Comment by baskren** [comment]
## CORECTION

 I am _finally_ able to get VS2022 for Mac (v17.0 - build 8989) working by setting the `MSBuildEnableWorkloadResolver = false` environment variable. 

For those who have the same issue:  Note that the most systemic way to set environment variables in MacOS is through `launchctl`:

```
 % launchctl setenv MSBuildEnableWorkloadResolver false
```

The old approach of systemically setting environmental variables was via `~/.MacOSX/environment.plist `.  However, that approach was removed in OS X Lion.

If you turn on **Diagnostic** log verbosity, the first thing listed in the build log will be the environment variables used for the build.   Setting environment variables in the traditional Unix way (`.profile`, `.bash_profile`, `.zprofile`, `.zshenv`, etc) will not work for applications.  

## TO THE MAINTAINERS

I still ask that you consider reopening this issue.  Several times it is stated that setting the `MSBuildEnableWorkloadResolver` environment variable is not necessary with more recent releases of VS2022.  My experience has proven that this is not universally true.

**Comment by marcpopMSFT** [comment]
@mrward the latest customer reply above is specific to VSMac 17.0. Have you seen this behavior in your own testing of VSMac 17?  Since you manage how the SDK gets inserted and configured into VSMac, I wonder if there is an issue with that specific scenario.

**Comment by mrward** [comment]
@marcpopMSFT There is this feedback ticket - https://developercommunity.visualstudio.com/t/Unable-to-find-SDK-’MicrosoftNETSDKWo/10053602 - which has the same error, and needs further investigation.

With this feedback ticket the problem seems to be that one of the project's in the solution is a non-sdk project (or one that targets .NET Framework v4) which causes VS Mac 17.0 to use MSBuild on mono for the solution, and .NET 6.0.300 SDK was the only SDK installed, which is not supported by MSBuild on mono. VS Mac downgrades the .NET SDK to allow MSBuild on mono to work, and then that means there is no SDK available which I suspect then causes the workload resolver to fail. This particular problem is specific to VS Mac and is not a dotnet sdk or runtime issue. dotnet restore/build would not be affected by this.

Possible workarounds that may fix this are to install .NET 6.0.10x SDK, or to uncheck the Solution options - Build - General - Build with MSBuild on Mono check box.

**Comment by baskren** [comment]
@mrward :  FWIW:  
- Among the things I tried were manually installing .NET 6.0.100 and .NET 6.0.300 SDKs.  
- My solution(s) are all UNO platform libraries and platform projects.  As such, they only target one of the following:
    - platform specific (for the platform executable projects)
    - .NetStandard 2.0
    - or the following:

```
    <TargetFrameworks>uap10.0.18362;netstandard2.0;Xamarin.iOS10;Xamarin.Mac20;MonoAndroid11.0;</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.vsmac')) ">netstandard2.0;Xamarin.iOS10;Xamarin.Mac20;MonoAndroid11.0;</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.Droid')) ">MonoAndroid11.0</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.iOS')) ">Xamarin.iOS10</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.macOS')) ">Xamarin.Mac20</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.GtK')) ">netstandard2.0</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.Tizen')) ">Tizen50</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.WPF')) ">netstandard2.0</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.UWP')) ">uap10.0.18362</TargetFrameworks>
    <TargetFrameworks Condition=" $(SolutionName.Contains('.Wasm')) ">netstandard2.0</TargetFrameworks>
```

 I use the above conditionals (and, as needed, corresponding solution files) to reduce build times when doing application debugging.

**Comment by praeclarum** [comment]
I am running into this with my VS for Mac 2019 and 2022 projects. It seems including a .NET 6 project into a solution breaks resolution. Hoping for a fix as this affects all my projects.

**Comment by tompi** [comment]
Thank you @baskren !

'''
launchctl setenv MSBuildEnableWorkloadResolver false
'''

finally brought back my rider ide

**Comment by jwoeste** [comment]
Just had this issue on VS 17.3.6 (build 20) on macOS Monterey 12.6

Wasted days before finding this "launchctl setenv MSBuildEnableWorkloadResolver false"

The issue started after installing latest Xamarin.iOS.

Seems like a lot of regression testing is left with the community.

**Comment by trampster** [comment]
I just had this issue with Visual Studio for Mac 17.3.8 (build 5). This issue is clearly not fixed and should be reopened.

**Comment by emmanuelpare** [comment]
I got this issue for the first time today with the latest Macos version 17.5

Workaround is to executer:
launchctl setenv MSBuildEnableWorkloadResolver false

Release ID: 1705000437
Git revision: a5d0fb913c1ff781c80fe725a3e0511c34e26a08
Build date: 2022-11-02 21:34:22+00
Build branch: release-17.5

INFO [2022-11-08 15:30:51Z]: Running on .NET 7.0.0 (64-bit)
INFO [2022-11-08 15:30:51Z]: Operating System: Mac OS X 13.0.0
Darwin 22.1.0 Darwin Kernel Version 22.1.0
    Sun Oct  9 20:15:09 PDT 2022
    root:xnu-8792.41.9~2/RELEASE_ARM64_T6000 arm64

**Comment by JonBushby77** [comment]
Got this issue today after updating to VS Mac 17.5. I need to set the environment variable every time i reboot.

**Comment by zbecknell** [comment]
Just adding to the anecdotes -- I also encountered the issue and had to set the environment variable.

**Comment by dimdrandakis** [comment]
Same here VS Mac 17.5 Preview (17.5 build 437)

**Comment by mrward** [comment]
The 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' not being found error can happen in VS Mac for different reasons. There is a recent feedback ticket where if there were old pre-release versions of .NET 7 installed then this could also happen:

https://developercommunity.visualstudio.com/t/After-updating-to-net-7-Visual-Studio-/10194756

The underlying bug, old workload was renamed in later .NET 7 SDK, was also reported for Windows - https://github.com/dotnet/sdk/issues/28947

The workaround there is to remove .NET completely and install it again. The developer community ticket gives information on how to do that.

If that does not solve the problem, then we would need the IDE log. Since the IDE log may contain personal data it would better if this was reported on [Developer Community](https://developercommunity.visualstudio.com/VisualStudioMac) or via Report a Problem from with VS Mac itself. Thanks.
